### PR TITLE
add default exports to all packages

### PR DESCRIPTION
**Pull request by toloudis**
We are using Jest to do unit tests on a project that is newly integrating with some code that imports zarrita.
Jest simply can not resolve zarrita imports and we determined that it's because jest can't understand the "export" construct in the package.jsons of the zarrita packages.

We found that just adding a "default" entry suddenly made everything work.
We are unsure if this would break anything else but it seems harmless and allows the module resolution to happen correctly at least in Jest.

### CODE REVIEW

**Comment by manzt** [comment]
Hi there, sorry you are having issues with unit tests, but prior to making upstream changes for a single library I'd like to see if there are alternatives that can suit your choice of using Jest. I assume you are using NodeJS, which version?

The `import` is a standard field and makes it explicit that zarrita is distributed only as ESM (not commonjs). I believe that adding a `default` could "trick" node into trying to `require` zarrita.js, which should not happen because the library is not intended to be consumed that way.

Would you share more context about the project or any relevant code? Some questions I'd like to consider first:

- Does your project's package.json use `"type": "module"`? One issue is that Jest might fallback to node's resolution, and without type module Node resolves to `require` exports, not `import`. the `type` field determines how Node resolves `.js` files are interpreted. I typically set `"type": "module"`. If you have any scripts that use require, you can change them to `.cjs` explicitly.

- Sometimes changing `type: module` can break other things if you are relying on commonjs behavior somewhere else. You can "opt-in" to ESM resolution for your tests with `NODE_OPTIONS`. ESM support in node is now stable, but the flag stil says experimental.

```json
  "scripts": {
    "test": "NODE_OPTIONS=--experimental-vm-modules jest"
  }
```

- A last option is that many package managers allow for rewriting package metadata. For example, `pnpm` you could apply the patch you have suggested here with a [`.pnpmfile.cjs`](https://pnpm.io/pnpmfile).

TL;DR - I'm not totally opposed to these changes, I'd just like to explore options for he

**Comment by toloudis** [comment]
Thanks for the thorough response.  

We're using node 20.  
To get more specific, we are bringing https://github.com/allen-cell-animated/volume-viewer into https://github.com/simularium/simularium-viewer . The latter package is using Jest.

I'll try to look into all the things you suggested and reply here.  I can confirm that simularium-viewer doesn't use type:"module" yet (but volume-viewer does)...

**Comment by toloudis** [comment]
Well we went for it and got rid of Jest in favor of vitest and everything is working.  I am fine to close this PR.  Thanks again for the info and for the great library.
